### PR TITLE
Add anomaly task

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Summary

- Copies Anomaly Classification to Anomaly task as requested in CVS-147611

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — COMMENTED** [review]
Can't we get rid of `OTXTaskType.ANOMALY_CLASSIFICATION` right away?
If it test passes, I'll review it again.

**Comment by ashwinvaidya17** [comment]
> Can't we get rid of `OTXTaskType.ANOMALY_CLASSIFICATION` right away? If it test passes, I'll review it again.

Yeah let's do that. Since OTX 2.x is not in Geti yet we can still do this

**Comment by samet-akcay** [comment]
I think it would be better to create a branch from `1.6.3`, to which we could merge this PR. We could then use that branch to release 1.6.4 or 1.7.0

File: src/otx/core/types/task.py [hunk 1]
```diff
@@ -15,7 +15,8 @@ class OTXTaskType(str, Enum):
     ACTION_CLASSIFICATION = "ACTION_CLASSIFICATION"
 
     # Anomaly Detection
-    ANOMALY_CLASSIFICATION = "ANOMALY_CLASSIFICATION"
```
Comment (jihyeonyi) [inline src/otx/core/types/task.py:18]: As I know, both ANOMALY_CLASSIFICATION and ANOMALY should exist in phase 1 of the anomaly task reduction. Please don't delete this for Geti.

Comment (jihyeonyi) [reply src/otx/core/types/task.py:18]: This comment is for OTX 1.x only.

**Comment by harimkang** [comment]
What is a clear target version of this PR?

**Comment by ashwinvaidya17** [comment]
> What is a clear target version of this PR?

Do you mean in which 2.x version? If so, I don't know. It is just to have parity with the 1.6.x version. So when we upgrade in Geti, it does not break.

**Comment by jihyeonyi** [comment]
> > What is a clear target version of this PR?
> 
> Do you mean in which 2.x version? If so, I don't know. It is just to have parity with the 1.6.x version. So when we upgrade in Geti, it does not break.

Hi, to be used in Geti, both 'ANOMALY' and 'ANOMALY_CLASSIFIACION' are expected in OTX 1.6.x. So, please don't delete `ANOMALY_CLASSIFICATION` from `TaskType` in otx.api.entities.model_template . Or do you have a plan to make a function for name conversion while integrating OTX to Geti?

**Comment by ashwinvaidya17** [comment]
> > > What is a clear target version of this PR?
> > 
> > 
> > Do you mean in which 2.x version? If so, I don't know. It is just to have parity with the 1.6.x version. So when we upgrade in Geti, it does not break.
> 
> Hi, to be used in Geti, both 'ANOMALY' and 'ANOMALY_CLASSIFIACION' are expected in OTX 1.6.x. So, please don't delete `ANOMALY_CLASSIFICATION` from `TaskType` in otx.api.entities.model_template . Or do you have a plan to make a function for name conversion while integrating OTX to Geti?

I believe that the plan is to just keep the classification task. Once we have greenlight from the backend team, we can remove the rest of the task. This PR https://github.com/openvinotoolkit/training_extensions/pull/3832 targets 1.6.x that is going to be added to Geti as of now. I believe based on this, we will come to know if we can remove the rest of the tasks. This PR is just to make Anomaly Classification -> Anomaly changes in 2.x

**Comment by jihyeonyi** [comment]
> hat is going to be added to Geti as of now
Oh there's another PR for that. Thanks for letting me know this. I got confused because the link commented on the JIRA ticket was for this PR (#3818).

**Comment by jihyeonyi** [comment]
> > > > What is a clear target version of this PR?
> > > 
> > > 
> > > Do you mean in which 2.x version? If so, I don't know. It is just to have parity with the 1.6.x version. So when we upgrade in Geti, it does not break.
> > 
> > 
> > Hi, to be used in Geti, both 'ANOMALY' and 'ANOMALY_CLASSIFIACION' are expected in OTX 1.6.x. So, please don't delete `ANOMALY_CLASSIFICATION` from `TaskType` in otx.api.entities.model_template . Or do you have a plan to make a function for name conversion while integrating OTX to Geti?
> 
> I believe that the plan is to just keep the classification task. Once we have greenlight from the backend team, we can remove the rest of the task. This PR #3832 targets 1.6.x that is going to be added to Geti as of now. I believe based on this, we will come to know if we can remove the rest of the tasks. This PR is just to make Anomaly Classification -> Anomaly changes in 2.x

Oh there's another PR for that. Thanks for letting me know this. I got confused because the link commented on the JIRA ticket was for this PR (https://github.com/openvinotoolkit/training_extensions/pull/3818).

**Comment by ashwinvaidya17** [comment]
Ah right yeah! Originally we were told that 2.x will be used so this PR was created then they said 1.6.x. So basically we have to PRs for two different versions.

**Review by samet-akcay — DISMISSED** [review]
lgtm, thanks for handling this!

**Comment by jihyeonyi** [comment]
Because #3832 is closed, I think ANOMALY_CLASSIFICATION should be alive here. 
Or could we make a separate branch for the alias?

**Comment by ashwinvaidya17** [comment]
Closing as I'll create a new PR for release 2.2

**Pull request closed without merging**